apps/test-suite/tests/ScreenOrientation.js

```diff
@@ -0,0 +1,296 @@
+'use strict';
+
+import { ScreenOrientation } from 'expo';
+import { Platform } from 'react-native';
+import { waitFor } from './helpers';
+
+export const name = 'ScreenOrientation';
+
+// run callback every interval until it returns something truthy
+const pollUntilSuccessAsync = async (callback, interval = 500) => {
```

Comment:
IMO it'd be preferable for implement this without polling and instead rely on events. The polling doesn't timeout and if tests are killed by Jasmine's timeout, this polling code will still run in the background indefinitely. (Perhaps we may want a small number of tests with safe polling in case the events break, but polling is always either too early or too late.) 

Instead of polling, we could structure this like:

```
// set up listener
add listener {
  if event.orientation !== desiredOrientation
    return
  if desiredOrientationLock !== null && event.orientationLock !== desiredOrientationLock
    return
  // yay, satisified
  remove listener
  cancel timeout
  resolve promise
}

// check current value
currentOrientation, currentOrientationLock = await [getOrientation(), getOrientationLock()]
  if currentOrientation !== desiredOrientation
    not satisfied
  if desiredOrientationLock !== null && currentOrientationLock !== desiredOrientationLock
    not satisfied
  // if we are satisfied
  remove listener
  resolve promise

// set up timeout
setTimeout {
  remove listener
  reject promise
}
```